1.1.0 Metadata functions for Microsoft SQL Server

splatteredbits opened the pull request.

splatteredbits committed:
Creating changelog.

splatteredbits committed:
Migrating Invoke-YSqlServerCommand tests to Pester 5.

splatteredbits committed:
Invoke-YSqlServerCommand should be able to use someone else's connection object.

splatteredbits committed:
Renamed Invoke-YSqlServerCommand to Invoke-YMSSqlCommand.

splatteredbits committed:
Adding NOTICE to repo and package. Using .txt file extensions for LICENSE and NOTICE files for easy double-clicking.

splatteredbits committed:
Creating Microsoft SQL Server metadata functions.

splatteredbits committed:
Fixed path to tests directory.

splatteredbits committed:
Creating Microsoft SQL Server metadata functions.

splatteredbits committed:
Migrating tests to Pester 5.

splatteredbits committed:
Adding missing docs.

splatteredbits committed:
Updating to use latest build patterns.

splatteredbits committed:
Renaming master → main.

decoyjoe commented on Tests/Test-YMsSqlExtendedProperty.Tests.ps1:
Empty file?

decoyjoe commented on Yodel/Functions/Invoke-YDbCommand.ps1:
```diff
@@ -67,45 +82,57 @@ function Invoke-YDbCommand
-        [switch]$AsScalar,
-        
+        # Return the result as a single value instead of a row.  If the command returns multiple rows/columns, the value
+        # of the first row's first column is returned.
```
Nit: Parameter docs should be above the `[Parameter()]` attribute.

decoyjoe commented on Yodel/Functions/MsSql/Initialize-YMsSqlDatabase.ps1:
```diff
@@ -0,0 +1,43 @@
+{
+    <#
+    .SYNOPSIS
+    Esnures a Microsoft SQL Server database exists.
```
Typo: `Esnures`

decoyjoe commented on Yodel/Functions/MsSql/Initialize-YMsSqlDatabase.ps1:
```diff
@@ -0,0 +1,43 @@
+    .DESCRIPTION
+    The `Initialize-YMsSqlDatabase` function creates a database in Microsoft SQL Server if it doesn't already exist.
+    Pass the connection to SQL Server to the `Connection` parameter and the name of the database to the `Name`
+    parameter. If the database doesn't exist (i.e. there's no record for it in `sys.databases`), it is created.
```
From these docs, I can't figure out how I would know if the function actually did anything or not. E.g. is anything returned? A message written?

splatteredbits committed:
Fixed: importing fails when module files merged together.

splatteredbits committed:
Layout

splatteredbits committed:
Docs

splatteredbits committed:
Fixed: importing fails when module files merged together.

splatteredbits committed:
Adding Get-YMsSqlTableColumn metadata function.

splatteredbits committed:
Adding Test-YMsSqlTableColumn function.

splatteredbits committed:
Missing tests for Test-YMsSqlExtendedProperty.

splatteredbits committed:
Allow users to reduce round trips to SQL Server by returning object metadata if it exists.

splatteredbits commented:
Came by to add another metadata function here and discovered this PR that had never been merged. Oops. 

Added two new functions `Get-YMsSqlTableColumn` and `Test-YMsSqlTableColumn`. Added `-PassThru` switch to all the `Test-*` functions. Added some missing tests.

@decoyjoe Ready for another review.

splatteredbits committed:
Prerelease version number not auto-incrementing.

decoyjoe commented on Yodel/Functions/MsSql/Get-YMsSqlTableColumn.ps1:
```diff
@@ -0,0 +1,87 @@
+        [Parameter(Mandatory)]
+        [String] $TableName,
+
+        [String] $Name
```
Missing parameter doc.

decoyjoe commented on Yodel/Functions/MsSql/Test-YMsSqlExtendedProperty.ps1:
```diff
@@ -0,0 +1,140 @@
+        [String] $SchemaName,
+
+        # The table name whose extended property to set, or the table name of the column whose extended property to
+        # set.
```
Should be "test" not "set".

decoyjoe commented on Yodel/Functions/MsSql/Test-YMsSqlExtendedProperty.ps1:
```diff
@@ -0,0 +1,140 @@
+        [Parameter(Mandatory, ParameterSetName='ForTableColumn')]
+        [String] $TableName,
+
+        # The column name whose extended property to set.
```
Should be "test" not "set".

decoyjoe commented on Yodel/Functions/MsSql/Test-YMsSqlTable.ps1:
```diff
@@ -0,0 +1,46 @@
+        [Parameter(Mandatory)]
+        [Data.Common.DbConnection] $Connection,
+
+        [String] $SchemaName = 'dbo',
+
+        [Parameter(Mandatory)]
+        [String] $Name,
```
Missing parameter docs.

decoyjoe commented on Yodel/Functions/MsSql/Test-YMsSqlTableColumn.ps1:
```diff
@@ -0,0 +1,58 @@
+
+    .DESCRIPTION
+    The `Test-YMsSqlTableColumn` function tests if a column exists on a table in SQL Server. Pass the connection to SQL
+    Server to the `Connection` parameter, the table name to the `TabkeName` parameter, and the column name to the
```
Typo on `TabkeName`

splatteredbits committed:
Fixing documentation.
